integration-spring/src/main/java/org/ocpsoft/rewrite/spring/SpringServiceEnricher.java

```diff
@@ -18,24 +18,34 @@
 import java.util.ArrayList;
 import java.util.Collection;
 
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+
 import org.ocpsoft.common.spi.ServiceEnricher;
 import org.ocpsoft.logging.Logger;
+import org.ocpsoft.rewrite.servlet.spi.ContextListener;
 import org.springframework.web.context.support.SpringBeanAutowiringSupport;
 
 /**
  * {@link ServiceEnricher} implementation for Spring.
  * 
  * @author Christian Kaltepoth
  */
-public class SpringServiceEnricher implements ServiceEnricher
+public class SpringServiceEnricher implements ServiceEnricher, ContextListener
```

Comment:
With this setup, the `SpringServiceEnricher` will be instantiated twice, won't it? Once as `ServiceEnricher` and once as `ContextListener`

Reply:
Yes, but only one of them would be used as an Enricher. The other would just handle the static. I could split this into a different ContextListener implementation, and try to deal with the EAR / shared lib classloader issue there.